### PR TITLE
[DYN-4517] TuneUp will fail to register if not closed between graph runs

### PR DESCRIPTION
**Pull request by ivaylo-matov**
PR aims to address https://jira.autodesk.com/browse/DYN-4517.
It also addresses 1. of https://autodesk.slack.com/archives/CPJ5UQW0Z/p1719284511762389 : TuneUp to not impact performance when not being actively used.

DYN-4517_Fix

- ViewModel.IsTuneUpChecked tracks state of MenuItem.IsChecked
- subscribes to CurrentWorkspace events only when MenuItem.IsChecked
- RunTime switches to Manual mode when MenuItem.IsChecked
- Branch based on https://github.com/DynamoDS/TuneUp/pull/41

FYI
@QilongTang 
@Ashish Ganta
@Amoursol

### CODE REVIEW

File: TuneUp/TuneUp.csproj [hunk 1]
```diff
@@ -1,23 +1,21 @@
...
-    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <TargetFramework>net8.0-windows</TargetFramework>
+	<UseWPF>true</UseWPF>
     <RootNamespace>TuneUp</RootNamespace>
     <AssemblyName>TuneUp</AssemblyName>
-    <TargetFrameworkVersion>v4.8</TargetFrameworkVersion>
+    <OutputType>Library</OutputType>
     <StartAction>Program</StartAction>
-    <StartProgram>C:\Program Files\Dynamo\Dynamo Core\2\DynamoSandbox.exe</StartProgram>
+    <StartProgram>C:\Program Files\Dynamo\Dynamo Core\3\DynamoSandbox.exe</StartProgram>
```
Comment (QilongTang) [inline TuneUp/TuneUp.csproj:9]: This is an example path right? Just curious

File: TuneUp/TuneUp.csproj [hunk 2]
```diff
@@ -64,34 +62,31 @@
...
-  <!-- To modify your build process, add your task inside one of the targets below and uncomment it. 
-       Other similar extension points exist, see Microsoft.Common.targets.
-  <Target Name="BeforeBuild">
-  </Target>
-  <Target Name="AfterBuild">
-  </Target>
-  -->
+  <!-- Including Custom Properties, PreBuildEvent, and Import -->
   <PropertyGroup>
     <DynamoVersion>2.12</DynamoVersion>
```
Comment (QilongTang) [inline TuneUp/TuneUp.csproj:91]: Do we need to update this?

File: TuneUp/TuneUpViewExtension.cs [hunk 3]
```diff
@@ -44,17 +44,26 @@ public override void Loaded(ViewLoadedParams p)
...
+            TuneUpMenuItem.Unchecked += (sender, args) => ViewModel.IsTuneUpChecked = false;
+
             // Add this view extension's menu item to the Extensions tab or View tab accordingly.
             var dynamoMenuItems = p.dynamoMenu.Items.OfType<MenuItem>();
-            var extensionsMenuItem = dynamoMenuItems.Where(item => item.Header.ToString() == "_Extensions");
+            // TODO: Investigate why TuneUpMenuItem is not added to the specified MenuItem
+            // When _Extensions is specified, TuneUpMenuItem goes to the View menu in Dynamo 3.0-
+            // and does not appear in Dynamo 3.1+.
+            // We need to specify _View for TuneUpMenuItem to be added to the Extensions menu
+            var extensionsMenuItem = dynamoMenuItems.Where(item => item.Header.ToString() == "_View");
```
Comment (QilongTang) [inline TuneUp/TuneUpViewExtension.cs:66]: Confused by this a bit.. we specify `_View` to be added to Extentions munue?

File: TuneUp/TuneUp.csproj (hunk 1 above)
Comment (ivaylo-matov) [reply TuneUp/TuneUp.csproj:9]: Yes

File: TuneUp/TuneUpViewExtension.cs (hunk 3 above)
Comment (ivaylo-matov) [reply TuneUp/TuneUpViewExtension.cs:66]: You and me, brother 😁 
I could not figure out why that is, might have to do with the way Dynamo loads the menus. Looking and the old master seem like this hasn't worked before either. In old demos Show TuneUp is always called from the View menu although the code was specifying `_Extensions`. I thought this is decent workaround for the time being as it's not a part of the ticket.

Comment (Amoursol) [reply TuneUp/TuneUpViewExtension.cs:66]: Two things: 
1. Thank you Ivo for shifting it to Extensions. This is 100% where it should live!
2. We should make sure the coat is hardened, so if there is a proper way to do it please refactor :)

**Comment by reddyashish** [comment]
@ivaylo-matov there are still 2 conflicts.

**Comment by ivaylo-matov** [comment]
@reddyashish , sorry. Tried to resolve them before a couple of times, but wasn't sure if I should Commit Merge.

**Review by reddyashish — APPROVED** [review]

File: TuneUp/TuneUp.csproj (hunk 2 above)
Comment (reddyashish) [reply TuneUp/TuneUp.csproj]: This was updated in the previous PR

**Comment by reddyashish** [comment]
LGTM. Thank you @ivaylo-matov.

**Comment by reddyashish** [comment]
@QilongTang Will open a new PR to publish the tuneup package.

**Pull request merged**